src/karts/abstract_kart.cpp

```diff
@@ -79,6 +99,12 @@ void AbstractKart::reset()
 }   // reset
 
 // ----------------------------------------------------------------------------
+const AbstractCharacteristic* AbstractKart::getCharacteristic() const
+{
+    return &(*m_characteristic);
```

Comment:
I do not understand, it is unique_ptr in your code https://github.com/Flakebi/stk-code/blob/kart-properties/src%2Fkarts%2Fabstract_kart.hpp#L81. I am missing something?
What you are basically doing here is getting the value pointed by the pointer then getting the address of the value, which is basically the pointer, what .get() does for you.
And I think it's safe to call .get on an unique_ptr, as long as the other parts of the code base just use it to call other methods, and only one class manages it's lifetime.